### PR TITLE
Add `:hover` for points and polygons

### PR DESCRIPTION
**Pull request by a-chabin**
- [x] Created `useMapHoverObject`  with `mousemove`, `mouseleave` handlers
- [x] Add hover styles
   - [x] Added `getLayerActiveStyle(normal, active)` for hover states
   - [x] Added autogenerate id to `<Source>` by `generateId` prop
- [x] Added hovers to layers
   - [x] OKNs (+ refactor)
   - [x] DTPs
   - [x] Red/purple/blue lines

https://github.com/ekaterinburgdev/map/assets/22644149/219a9c42-f41f-4c25-9afd-0e1c44cf3e76

### CODE REVIEW

File: components/UI/Map/VectorMap/OknSource.tsx [hunk 1]
```diff
@@ -1,52 +1,91 @@
...
+    },
+    security: {
+        data: '/ekb-okn-security.json',
+        layer: 'ekb-okn-security-layer',
+        source: 'ekb-okn-security-source',
+        zone: OknAreaType.SecurityZone,
+    },
+    objects: {
+        data: '/ekb-okn-objects.json',
+        layer: 'ekb-okn-objects-layer',
```
Comment (gcor) [inline components/UI/Map/VectorMap/OknSource.tsx:35]: lets rename to layerId, sourceIds, dataPath?)

**Review by gcor — APPROVED** [review]

File: components/UI/Map/VectorMap/DtpSource.tsx [hunk 2]
```diff
@@ -62,7 +66,7 @@ export function DtpSource() {
         type: 'circle',
         source: 'ekb-dtp-source',
         paint: {
-            'circle-radius': 8,
+            'circle-radius': getLayerActiveStyle<number>(8, 10),
```
Comment (gcor) [inline components/UI/Map/VectorMap/DtpSource.tsx:69]: probably better to use objects in the args of getLayerActiveStyle<number>({initial: 8, active: 10}). Just for clarity

Comment (gcor) [reply components/UI/Map/VectorMap/DtpSource.tsx:69]: oh. and if you set default initial value, we can rename `getLayerActiveStyle` to `getLayerStyle`

Comment (a-chabin) [reply components/UI/Map/VectorMap/DtpSource.tsx]: Good. I renamed.

In the future we will make `{initial, hover, active}`

File: components/UI/Map/VectorMap/OknSource.tsx (hunk 1 above)
Comment (a-chabin) [reply components/UI/Map/VectorMap/OknSource.tsx]: Done

**Pull request merged**